frontend/src/routes/admin/AddProductsDialog.svelte

```diff
@@ -0,0 +1,275 @@
+<script lang="ts">
+	import * as Form from '$lib/components/ui/form';
+	import * as Dialog from '$lib/components/ui/dialog';
+	import * as DropdownMenu from '$lib/components/ui/dropdown-menu';
+	import { Input } from '$lib/components/ui/input';
+	import { Textarea } from '$lib/components/ui/textarea';
+	import { Button } from '@/components/ui/button';
+	import { addProductSchema, type AddProductSchema } from './schema';
+	import { type SuperValidated, type Infer, superForm } from 'sveltekit-superforms';
+	import { zodClient } from 'sveltekit-superforms/adapters';
+	import AddIcon from 'virtual:icons/mdi/add';
+	import DeleteIcon from 'virtual:icons/mdi/delete-forever-outline';
+	import SaveIcon from 'virtual:icons/mdi/content-save-outline';
+	import CloseIcon from 'virtual:icons/mdi/close';
+	import dndIcon from '$lib/assets/admin/icons/dndIcon.svg';
+	import { Control } from 'formsnap';
+
+	// export let data: SuperValidated<Infer<AddProductSchema>>;
+	let { data } = $props();
+
+	const addProductForm = superForm(data, {
+		validators: zodClient(addProductSchema)
+	});
+
+	let isHoveredMainImage = $state(false);
+
+	let categories = [
+		{ value: 'merch', label: 'Merchandise', color: '#4D1078' },
+		{ value: 'cafe', label: 'Cafe & Pastries', color: '#FB7A4F' },
+		{ value: 'print', label: "Misbeek's Printing", color: '#ffabff' },
+		{ value: 'minimart', label: 'Mini-mart', color: '#F9F871' },
+		{ value: 'workshop', label: 'Workshop', color: '#32beaf' }
+	];
+	let darkTextCategories = ['print', 'minimart'];
+	let categoryValue = $state('');
+	const triggerCategory = $derived(
+		categories.find((c) => c.value === categoryValue)?.label ?? 'Select Category'
+	);
+
+	// NOTE: The creator's names should be fetched from the backend (username)
+	let creators = [
+		{ uuid: '1', name: 'admin1', color: '#ee1768' },
+		{ uuid: '2', name: 'admin2', color: '#5eb5e3' },
+		{ uuid: '3', name: 'admin3', color: '#b0d253' }
+	];
+	let creatorsValue = $state([]) as string[];
+
+	function isCreatorIncluded(uuid: string) {
+		return creatorsValue.includes(uuid);
+	}
+	function toggleCreator(uuid: string) {
+		if (creatorsValue.includes(uuid)) {
+			// remove from list
+			creatorsValue = creatorsValue.filter((c) => c !== uuid);
+		} else {
+			// add to list
+			creatorsValue = [...creatorsValue, uuid];
```

Comment:
Refactor to use `push(uuid)`[^1] instead of iterating over creatorsValue.

[^1]: https://svelte.dev/docs/svelte/$state#Deep-state